crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -610,7 +620,9 @@ where
     // If there is no path to this block from the tip due to snap sync, we'll start archiving from
     // an earlier segment, then start mapping again once archiving reaches this block.
     if let Some(block_number) = create_object_mappings.block() {
-        best_block_to_archive = best_block_to_archive.min(block_number);
+        // There aren't any mappings in the genesis block, so starting there is pointless.
+        // (And causes errors on restart, because genesis block data is pruned.)
+        best_block_to_archive = best_block_to_archive.min(block_number).max(1);
```

Comment:
It will work fine with `--sync full`, maybe make the input `NonZero` if zero doesn't make sense as a value?

Reply:
Yeah it’s kind of a weird case. There’s nothing wrong with having mappings in the genesis block, it’s just something we choose not to do.

I’m also not sure if we can get a zero best_block_to_archive here, I’ll check the code again.

Reply:
The non-zero change is now PR #3247

Reply:
We can get a zero best_block_to_archive because we do a saturating subtraction on the best block, so we need to keep the max(1). (PR #3247 doesn’t fix that, it only stops the config being zero.)